org.uqbar.project.wollok.lib/src/wollok.wlk

```diff
@@ -508,6 +524,28 @@ package lang {
 		
 		method abs() native
 		method invert() native
+		method gcd(other) native
+		method lcm(other) {
+			const mcd = self.gcd(other)
+			return self * (other / mcd)
+		}
+		method digits() {
+			var digits = self.toString().size()
+			if (self < 0) {
+				digits -= 1
+			}
+			return digits
+		}
+		method prime() {
+			if (self == 1) return false
+			var isPrime = true
+			(2..self - 1).forEach({ i =>
+				// Horrible definition, but return doesn't exit from loop
+				// and I need fold, any methods in Range
```

Comment:
The non local return + not having a native for construct is reducing the language expresivness.

Perhaps we should discuss adding non local returns like Smalltalk or Ruby's blocks. Otherwise there are some algorithims than can't be implementes natively.

Alternatively, de could implement a while method that implements such short circuit logic.